src/citrine/resources/design_space.py

```diff
@@ -55,33 +56,91 @@ def _validate_write_request(self, design_space: DesignSpace):
                 raise ValueError(msg.format(self._enumerated_cell_limit, width * length))
         return
 
-    def _hydrate_design_space(self, model: DesignSpace) -> List[DesignSpace]:
-        if model.typ != "ProductDesignSpace":
-            return model
+    def _hydrate_design_space(self, design_space: DesignSpace) -> List[DesignSpace]:
+        if design_space.typ != "ProductDesignSpace":
+            return design_space
 
         subspaces = []
-        for subspace in model.subspaces:
+        for subspace in design_space.subspaces:
             if isinstance(subspace, (str, UUID)):
                 warnings.warn("Support for UUIDs in subspaces is deprecated as of 2.16.0, and "
                               "will be dropped in 3.0. Please use DesignSpace objects instead.",
                               DeprecationWarning)
                 subspaces.append(self.get(subspace))
             else:
                 subspaces.append(subspace)
-        model.subspaces = subspaces
-        return model
+        design_space.subspaces = subspaces
+        return design_space
 
-    def register(self, model: DesignSpace) -> DesignSpace:
+    def register(self, design_space: DesignSpace) -> DesignSpace:
         """Create a new design space."""
-        self._validate_write_request(model)
-        hydrated_model = self._hydrate_design_space(model)
-        return AbstractModuleCollection.register(self, hydrated_model)
+        self._verify_write_request(design_space)
+        hydrated_ds = self._hydrate_design_space(design_space)
+
+        registered_ds = super().register(hydrated_ds)
+
+        # If the initial response is invalid, just return it.
+        # If not, kick off validation, since we never exposed saving a design space without
+        # validation so we should continue to do it automatically
+        if registered_ds.failed():
+            return registered_ds
+        else:
+            return self._validate(registered_ds.uid)
+
+    def update(self, design_space: DesignSpace) -> DesignSpace:
+        """Update and validate an existing DesignSpace."""
+        self._verify_write_request(design_space)
+        hydrated_ds = self._hydrate_design_space(design_space)
+        updated_ds = super().update(hydrated_ds)
+
+        # The /api/v3/design-spaces endpoint switched archiving from a field on the update payload
+        # to their own endpoints. To maintain backwards compatibility, all design spaces have an
+        # _archived field set by the archived property. It will be archived if True, and restored
+        # if False. It defaults to None, which does nothing. The value is reset afterwards.
+        if design_space._archived is True:
+            self.archive(design_space.uid)
```

Comment:
Philosophically, it enforces separation of concerns. The update endpoint can focus on changes to the data itself, whereas archive/restore can focus on the archive state.
It also allows the client to archive without needing to pass a copy of the resource. The client need only know the resource's ID.
And it's a natural way to pass that information to the backend. Previously, it was part of the payload. But in the new setup, since archive status is metadata, it's not part of the input payload. We could allow it to be specified as a query parameter, but that feels clunky as a way to direct action (which is part of the reason `validate` is now it's own endpoint).